proposals/0540-jsem.rst

```diff
@@ -171,18 +175,26 @@ know anything about the jobserver that has spawned them.
 Implementation of the protocol
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The implementation relies on an interface for system semaphores. For this, we
-propose to using the functionality provided by the ``unix`` and ``Win32``
-libraries. In practice, our implementation ships with a compatibility layer
-which provides an abstraction over those two libraries. This handles creation of
-fresh semaphores, acquisition, and release of tokens, including a mechanism for
-interrupting a wait on a semaphore when a token is no longer needed.
+The implementation is built on ``semaphore-compat``, a Haskell library that
+provides an abstract semaphore interface. Tools that wish to use this feature
+must do so by means of this library.
```

Comment:
So the API of the library is part of this proposal. Please can you lay out the API of the library, here?

Reply:
> So the API of the library is part of this proposal. Please can you lay out the API of the library, here?

You did not respond.... yet authors *must* use this API so it seems important to the proposal.  Am I off target?  This is far from what I know!

Reply:
@wz1000 can you add the core operations here (i.e. `create`, `acquire`, `release`) and link to https://hackage.haskell.org/package/semaphore-compat-1.0.0/docs/System-Semaphore.html for the details.

Ultimately the library may evolve and the API Docs should be authoritative, but it's good to capture the starting point here for posterity.